mypyc/genops.py

```diff
@@ -3495,6 +3594,44 @@ def instantiate_generator_class(self) -> Value:
         self.add(SetAttr(curr_env_reg, NEXT_LABEL_ATTR_NAME, zero_reg, fitem.line))
         return generator_reg
 
+    def add_raise_exception_blocks_to_generator_class(self, line: int) -> None:
+        """
+        Generates blocks to check if error flags are set while calling the helper method for
+        generator functions, and raises an exception if those flags are set.
+        """
+        cls = self.fn_info.generator_class
+
+        # Check to see if an exception was raised.
+        error_block = BasicBlock()
+        ok_block = BasicBlock()
+        comparison = self.binary_op(cls.exc_type_reg, self.none(), 'is not', line)
+        self.add_bool_branch(comparison, error_block, ok_block)
+
+        self.activate_block(error_block)
+
+        # Check to see if the traceback field is set.
+        if_tb_block = BasicBlock()
+        else_block = BasicBlock()
+        comparison = self.binary_op(cls.exc_exc_tb_reg, self.none(), 'is not', line)
+        self.add_bool_branch(comparison, if_tb_block, else_block)
+
+        # If the traceback field is set, then raise an exception and set the traceback field to the
+        # value that was passed in.
+        self.activate_block(if_tb_block)
+        self.primitive_op(raise_exception_with_tb_op,
+                          [cls.exc_type_reg, cls.exc_value_reg, cls.exc_exc_tb_reg], line)
+        self.goto(ok_block)
```

Comment:
Put in `Unreachable` instead probably, since it should never get past the raise exception. (It is basically a bug that we can't treat raising exceptions as terminating a block.)